common/src/chain/transaction/signature/mod.rs

```diff
@@ -192,8 +197,42 @@ fn verify_standard_input_signature<T: Transactable>(
     input_num: usize,
 ) -> Result<(), DestinationSigError> {
     let sighash = signature_hash(witness.sighash_type(), tx, inputs_utxos, input_num)?;
-    witness.verify_signature(chain_config, outpoint_destination, &sighash)?;
-    Ok(())
+    let input_utxo = inputs_utxos.get(input_num).ok_or(
+        DestinationSigError::InvalidInputUtxoIndex(input_num, inputs_utxos.len()),
+    )?;
+
+    let raw_signature = match input_utxo {
+        Some(input_utxo) => match input_utxo {
+            TxOutput::Transfer(_, _)
+            | TxOutput::LockThenTransfer(_, _, _)
+            | TxOutput::Burn(_)
+            | TxOutput::CreateStakePool(_, _)
+            | TxOutput::ProduceBlockFromStake(_, _)
+            | TxOutput::CreateDelegationId(_, _)
+            | TxOutput::DelegateStaking(_, _)
+            | TxOutput::IssueFungibleToken(_)
+            | TxOutput::IssueNft(_, _, _)
+            | TxOutput::DataDeposit(_) => witness.raw_signature().to_owned(),
+            TxOutput::AtomicSwap(_, _) => {
+                let htlc_spend =
+                    AuthorizedHashedTimelockContractSpend::from_data(witness.raw_signature())?;
+                match htlc_spend {
+                    AuthorizedHashedTimelockContractSpend::Secret(_, raw_signature) => {
+                        raw_signature
+                    }
+                    AuthorizedHashedTimelockContractSpend::Multisig(raw_signature) => raw_signature,
+                }
+            }
+        },
+        None => witness.raw_signature().to_owned(),
+    };
```

Comment:
This is wrong. Script execution should not be part of signature verification.

Reply:
but this is just signature extraction not verification

Reply:
It's "script execution". Shouldn't be done here.

Reply:
We're getting closer to this making sense.

I think the right way to go is to create something that matches the evaluation of the "script" in these outputs and their corresponding inputs.

We have half of that in the destination getter. We need something similar to "evalulate signature". Both together will yield something that may call the signature verification function in this module, which certainly doesn't need to know anything about htlcs. This logic has to be moved out of here. It's something that's used by the transaction verifier, but it doesn't have to be in the transaction verifier. It can be a submodule in there. We can put it in the transaction verifier as a first step.

Reply:
I will resolve this and continue at the next spot.